File: src/main/java/com/github/nsc/de/compiler/parser/node/functions/FunctionCallNode.java
package com.github.nsc.de.compiler.parser.node.functions;

import com.github.nsc.de.compiler.parser.node.ValuedNode;

import java.util.Arrays;

public class FunctionCallNode implements ValuedNode {

    private final ValuedNode function;
    private final ValuedNode[] args;


    public FunctionCallNode(ValuedNode function, ValuedNode[] args) {
        this.function = function;
        this.args = args;
    }

    public ValuedNode getFunction() {
		return function;
	}

    public ValuedNode[] getArgs() {
        return args;
    }

    @Override
    public String toString() {
        return "FunctionCallNode{" +
                "function='" + getFunction() + '\'' +
                ", args=" + Arrays.toString(args) +
                '}';
    }
}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/functions/FunctionCallNode.kt
package com.github.shakelang.shake.parser.node.functions

import com.github.shakelang.shake.parser.node.ValuedNode
import com.github.shakelang.shake.util.characterinput.position.PositionMap

class FunctionCallNode(map: PositionMap, val function: ValuedNode, val args: Array<ValuedNode>) : ValuedNode(map) {

    override fun toJson(): Map<String, *> =
        mapOf("name" to "FunctionCallNode", "function" to function.json, "args" to args.map { it.json })

}